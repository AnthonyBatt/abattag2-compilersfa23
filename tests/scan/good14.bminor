y = mx + b;

return (f-32)*5/9

if (yup)
{
	print "he he ha"
}
else 
{
	foo("he he ha")
}
for (integer i = 0; i < 100; i++)
{
	print "nope";
}
