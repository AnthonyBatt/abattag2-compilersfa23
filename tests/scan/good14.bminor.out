TOKEN_ID                  y
TOKEN_ASSIGNMENT          =
TOKEN_ID                  mx
TOKEN_PLUS                +
TOKEN_ID                  b
TOKEN_SEMICOLON           ;
TOKEN_RETURN              return
TOKEN_PAREN_OPEN          (
TOKEN_ID                  f
TOKEN_INTEGER_LITERAL     -32
TOKEN_PAREN_CLOSE         )
TOKEN_MULTI               *
TOKEN_INTEGER_LITERAL     5
TOKEN_DIVIDE              /
TOKEN_INTEGER_LITERAL     9
TOKEN_IF                  if
TOKEN_PAREN_OPEN          (
TOKEN_ID                  yup
TOKEN_PAREN_CLOSE         )
TOKEN_BRACE_OPEN          {
TOKEN_PRINT               print
TOKEN_STRING_LITERAL      he he ha
TOKEN_BRACE_CLOSE         }
TOKEN_ELSE                else
TOKEN_BRACE_OPEN          {
TOKEN_ID                  foo
TOKEN_PAREN_OPEN          (
TOKEN_STRING_LITERAL      he he ha
TOKEN_PAREN_CLOSE         )
TOKEN_BRACE_CLOSE         }
TOKEN_FOR                 for
TOKEN_PAREN_OPEN          (
TOKEN_INTEGER             integer
TOKEN_ID                  i
TOKEN_ASSIGNMENT          =
TOKEN_INTEGER_LITERAL     0
TOKEN_SEMICOLON           ;
TOKEN_ID                  i
TOKEN_LT                  <
TOKEN_INTEGER_LITERAL     100
TOKEN_SEMICOLON           ;
TOKEN_ID                  i
TOKEN_POST_INC            ++
TOKEN_PAREN_CLOSE         )
TOKEN_BRACE_OPEN          {
TOKEN_PRINT               print
TOKEN_STRING_LITERAL      nope
TOKEN_SEMICOLON           ;
TOKEN_BRACE_CLOSE         }
